### ISSUE TITLE
VAE Loss calculation

### ISSUE THREAD

**Issue by ortix** [cyrilibrahim/VAE_keras_spirale#1]
I am curious why you're calculating your VAE loss this way:
```
def vae_loss(y_true, y_pred):
    """ Calculate loss = reconstruction loss + KL loss for each data in minibatch """
    # E[log P(X|z)] here a Gaussian BCE
    recon = -K.log(2*np.pi) -  0.5 * K.sum(std_decoder,axis=1) - 0.5*K.sum((K.square(inputs - mu_decoder)/(K.exp(std_decoder))),axis=1)
    # D_KL(Q(z|X) || P(z|X)); calculate in closed form as both dist. are Gaussian
    kl_loss = 0.5 * ( K.sum(1 + log_sigma - K.square(mu) - K.exp(log_sigma), axis=1))
    return   (data_number/mini_batch_size) * -K.sum(alpha*kl_loss + recon,axis=0)
```

More specifically, I don't understand your reconstruction loss. I would appreciate a response as to why you're taking the expected value of the log posterior
